### ISSUE TITLE
Ubuntu Focal and Python 3.8 Wheel compatibility issue.

### ISSUE THREAD

**Issue by digitalsleuth** [area1/stpyv8#18]
ISSUE: 
- Import of STPyV8 in Python3.8.5 (comes with Ubuntu Focal 20.04 by default) results in two errors indicating static link to libboost1.65.1 and python3.6.

STEPS TO REPRODUCE: 
- I used a barebones Ubuntu Focal docker with only salt-minion, wget, and gnupg installed. 
- Installed libboost-python-dev, libboost-system-dev, and libboost-dev (which are all 1.71.0 by default on Focal). 
- Installed build-essential, python3, python3-pip and sudo (python3-dev came down automatically and was installed)
- python3 -m pip install https://github.com/area1/stpyv8/releases/download/v8.8.278.10/stpyv8-8.8.278.10-cp38-cp38-linux_x86_64.whl
- Run python3, import STPyV8

ERROR 1:
```
>>> import STPyV8
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/STPyV8.py", line 10, in <module>
    import _STPyV8
ImportError: libboost_system.so.1.65.1: cannot open shared object file: No such file or directory
```
In an effort to find the cause, or any additional errors, I attempted:
`ln -s /usr/lib/x86_64-linux-gnu/libboost_system.so.1.71.0 /usr/lib/x86_64-linux-gnu/libboost_system.so.1.65.1`

Then tried again and got ERROR 2:
```
root@04275e6d4020:/# python3
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import STPyV8
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/STPyV8.py", line 10, in <module>
    import _STPyV8
ImportError: libboost_python3-py36.so.1.65.1: cannot open shared object file: No such file or directory
```

Now, while I understand that the linking of the shared objects is not a reasonable workaround, it is helpful in fault-finding as it can help determine if there are also other requirements which are not met, as it the case here.

If STPyV8 is not intended to work on any version of libboost above 1.65.1, then ERROR 1 is expected behaviour, and I would be happy to close the issue. However, since python 3.8 is not available on an Ubuntu/Debian system **by default** where libboost 1.65.1 is available (Ubuntu 18.04 = 3.6, Debian 10 = 3.7), then I wanted to raise this issue in the event this is an oversight, especially in light of ERROR 2.

I am happy to assist in additional testing. Please let me know if you require any further information.

**Comment by buffer** [comment]
Thanks for the report. Unfortunately this issue is not so easy to fix. Currently the Github Actions build and release workflow is mainly focused on Ubuntu-18.04 because that's the platform I use for my day to day operations. 

I would be glad to support more effectively other platforms but there are some barriers. First of all building a _manylinux_ image is not possible at the moment as V8 compilation requires a very recent libc. 

The approach I would like to test is based on repacking the generated wheels including the libraries you mentioned. But again this is not straightforward as changing RPATH is required to allow the STPyV8 so to locate such libraries. Still don't know if that could be the way to go but I will keep on thinking about a feasible solution (and keep the ticket open as a reminder). 

Meanwhile a quick and dirty way to generate the wheel you need would be to clone the repo and just modify build-linux.yml _runs-on_ directive to _ubuntu-20.04_. 

Of course if you have suggestions about more general solutions they are really appreciated. Thanks!

**Comment by buffer** [comment]
FYI I am working on a new workflow to generate Python wheels for Ubuntu-20.04. The new workflow is still far from perfect but the first results/artifacts are available at [1]. It would be great if you could spend some cycles testing if the generated wheel works fine for you too. Thanks!

[1] https://github.com/area1/stpyv8/actions/runs/495530966.

**Comment by digitalsleuth** [comment]
Sounds good to me. I'll give it a try this afternoon once I get home. Thanks for following up on this. I'll let you know how it goes. 

Cheers!

**Comment by digitalsleuth** [comment]
Took me a little while, but I managed to get this to build and work on Ubuntu 20.04, and all tests passed. Installed and configured Thug to test as well but Thug failed with two errors (not STPyV8 related).

Tested in a 20.04 docker, installed packages as required (libboost-python-dev, libboost-system-dev, libboost-dev), then downloaded and installed the workflow wheel. No problems at all. The 20.04 workflow and wheel work great!
